sound/soc/sof/sof-audio.c

```diff
@@ -394,15 +395,41 @@ static int sof_free_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dap
  * The error path in this function ensures that all successfully set up widgets getting freed.
  */
 static int sof_set_up_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dapm_widget *widget,
-				      int dir, struct snd_soc_dapm_widget_list *list)
+				      int dir, struct snd_sof_pcm *spcm)
 {
+	struct snd_sof_pipeline_trigger_info *pipeline_info = &spcm->stream[dir].pipeline_info;
+	struct snd_soc_dapm_widget_list *list = spcm->stream[dir].list;
+	struct snd_sof_widget *swidget = widget->dobj.private;
 	struct snd_soc_dapm_path *p;
 	int ret;
 
-	if (widget->dobj.private) {
+	if (swidget) {
+		int i;
+
 		ret = sof_widget_setup(sdev, widget->dobj.private);
 		if (ret < 0)
 			return ret;
+
+		/*
+		 * Add the widget's pipe_widget to the list of pipelines to be triggered if not
+		 * already in the list. This will result in the pipelines getting added in the
+		 * order source to sink.
+		 */
+		for (i = 0 ; i < pipeline_info->count; i++)
+			if (pipeline_info->pipelines[i] == swidget->pipe_widget)
+				break;
+
+		if (i == pipeline_info->count) {
+			u32 pipeline_count = pipeline_info->count++;
```

Comment:
No need to define a new variable. Can use ```i``` directly.

Reply:
yes, let me do this in a fixup